pkg/utils/utils.go

```diff
@@ -169,3 +175,62 @@ func AppendPortIfNotExists(slice []v1.ContainerPort, elems ...v1.ContainerPort)
 	}
 	return slice
 }
+
+// IsCrdAvailable checks if a given CRD is present in the cluster by verifying the
+// existence of its API.
+func IsCrdAvailable(config *rest.Config, groupVersion, kind string) (bool, error) {
+	gvResources, err := GetAvailableResourcesForApi(config, groupVersion)
+	if err != nil {
+		return false, err
+	}
+
+	found := false
```

Comment:
this if could be reduced to:

```
return gvResources != nil && any(crd.Kind == kind for crd := range gvResources.APIResources), nil
```

Reply:
I copied and pasted and it does not build....
I guess this is pseudocode. If it is, then yes, what you provided is the short form of all that code.